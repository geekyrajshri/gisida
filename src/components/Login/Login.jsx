import React from 'react';
import Cookie from 'js-cookie';
import PropTypes from 'prop-types';

require('./Login.scss');


class Login extends React.Component {
  constructor(props) {
    super(props);
    this.state = { loginError: false };
  }

  handleLogin(password) {
    if (password === this.props.appConfig.password) {
      Cookie.set('dsauth', true);
      location.reload();
    } else {
      this.setState({ loginError: true });
    }
  }

  render() {
    return (
      <div className="login">
        <div>
          <form className="login-form" onSubmit={(e) => { e.preventDefault(); this.handleLogin(this.password.value); }}>
            <div className="form-group">
              <div className="brand-login" />
              { this.state.loginError === true ? <div className="alert alert-danger">Incorrect password.</div> : null }
              <label htmlFor="password">Enter your Password</label>
              <input className="form-control" type="password" ref={(input) => { this.password = input; }} autoFocus />
            </div>
            <button className="btn btn-default center-block" type="submit">Log In</button>
          </form>
        </div>
      </div>);
  }
}

export default Login;

Login.propTypes = {
  appConfig: PropTypes.string.isRequired,
};
