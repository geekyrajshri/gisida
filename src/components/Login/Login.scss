.login {
  width: 100%;
  background: white;
  color: black;
  height: 100%;
  position: fixed;

  .brand-login {
    display: block;
    width: 200px;
    height: 200px;
    text-align: center;
    color: #ccc;
    background-size: 70%;
    text-indent: -9999px;
  }

  .framework-brand-login {
    width: 320px;
    height: 120px;
    display: block;
    text-align: center;
    color: #ccc;
    text-indent: -9999px;
  }

  .password-label {
    display: inline-block;
    font-weight: bold;
    color: #333;
    width: 100%;
    text-align: center;
    margin: 10px auto;
  }

  .form-control {
    margin: 0 auto;
    width: 80%;
  }

  .login-form {
    position: fixed;
    top: 35%;
    left: 50%;
    transform: translate(-50%, -50%);

    .login input {
      background: rgba(0, 0, 0, 0.8);
    }
  }

  .funded-by {
    font-weight: bold;
    color: #333;
    text-align: center;
    position: absolute;
    bottom: 19%;
    left: 50%;
    transform: translate(-50%);
  }

  .donor-logos {
    position: absolute;
    bottom: 2%;
    left: 50%;
    transform: translate(-50%);

    .donors {
      display: inline-block;
      line-height: 60px;
      vertical-align: middle;
      padding: 5px;
    }

    img {
      position: relative;
      float: left;
      width: 90px;
      height: 110px;
    }
  }
}
