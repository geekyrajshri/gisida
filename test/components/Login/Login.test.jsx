import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../../../src/components/Login/Login';

describe('Login', () => {
    it('Login renders login screen', () => {
      const component = renderer.create(<Login appConfig={{password: "default"}}/>);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
});
