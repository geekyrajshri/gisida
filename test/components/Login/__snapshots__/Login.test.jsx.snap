// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Login renders login screen 1`] = `
<div
  className="login"
>
  <div>
    <form
      className="login-form"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <div
          className="brand-login"
          style={
            Object {
              "background": "url(undefined) no-repeat center center",
            }
          }
        />
        <label
          className="password-label"
          htmlFor="password"
        >
          Enter your password
        </label>
        <input
          autoFocus={true}
          className="form-control"
          type="password"
        />
      </div>
      <button
        className="btn btn-default center-block"
        type="submit"
      >
        Log In
      </button>
    </form>
  </div>
</div>
`;
